src/build/cargo.rs

```diff
@@ -405,7 +389,7 @@ impl Executor for RlsExecutor {
                 _ => {}
             }
         } else {
-            cmd.status().expect("Couldn't execute rustc");
+            cmd.exec()?;
```

Comment:
Just as a note, it does roughly the same what [this](https://github.com/rust-lang-nursery/rls/pull/456/files#diff-7946bdad71ef1f573a8d51890bda997fL320), minus returning from the function on success - we still store the cached args/envs at the end of the function